src/routes/PrivateRoutes.jsx

```diff
@@ -0,0 +1,11 @@
+import { useSelector } from 'react-redux';
+import { Navigate, Outlet, useLocation } from 'react-router-dom';
+
+const PrivateRoutes = () => {
+    const { token } = useSelector((state) => state.auth);
+    const location = useLocation();
+
+    return token ? <Outlet /> : <Navigate to="/" state={{ from: location }} replace />;
```

Comment:
This is something new I got to know. Need to get more on this `Outlet` concept.

Reply:
Basically it helps in rendering out the child route elements.